assets/src/edit-story/app/media/media3p/api/media3pApiProvider.js

```diff
@@ -47,10 +47,38 @@ const Providers = {
 function Media3pApiProvider({ children }) {
   const MEDIA_PAGE_SIZE = 20;
 
-  function constructFilter(provider, searchTerm, mediaType) {
+  /**
+   * Constructs the filter used for the listMedia requests.
+   *
+   * @param {Object} obj - An object with the options.
+   * @param {string} obj.provider The provider to get the media from.
+   * Currently only 'unsplash' is supported.
+   * @param {?string} obj.searchTerm Optional search term to send,
+   * eg: 'cute cats'.
+   * @param {?string} obj.selectedCategoryId Optional id of the selected
+   * category.
+   * @param {?string} obj.mediaType The media type of results to get.
+   * Currently ignored by the API as Unsplash only handles images.
+   * @return {string} A filter string.
+   */
+  function constructFilter({
+    provider,
+    searchTerm,
+    selectedCategoryId,
+    mediaType,
+  }) {
+    if (provider.toLowerCase() !== Providers.UNSPLASH) {
+      throw new Error(`Unsupported provider: ${provider}`);
+    }
+    if (selectedCategoryId && searchTerm) {
+      throw new Error(
+        `searchTerm and selectedCategoryId are mutually exclusive.`
+      );
+    }
     return [
-      provider ? `provider:${provider}` : null,
+      `provider:${provider}`,
       mediaType ? `type:${mediaType}` : null,
+      selectedCategoryId ? `category:${selectedCategoryId}` : null,
```

Comment:
[nit]: These can be written as:

mediaType && `type:${mediaType}`,
selectedCategoryId && `category:${selectedCategoryId}`,

Reply:
Done.